apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Rate.jsx

```diff
@@ -76,48 +84,62 @@ function Rate({ value, selected, onSelect, fromCurrency, toCurrency }: Props) {
-                  from={fromCurrency}
-                  to={toCurrency}
-                  rate={value.magnitudeAwareRate}
+            {centralized && (
+              <Box>
+                <Box style={{ height: "19.5px", justifyContent: "center" }}>
+                  <Price
+                    withEquality
+                    withIcon={false}
+                    from={fromCurrency}
+                    to={toCurrency}
+                    rate={value.magnitudeAwareRate}
+                    fontWeight="600"
+                  />
+                </Box>
+                <Text fontSize={3} color="palette.text.shade40">
+                  <Trans
+                    i18nKey={
+                      value.tradeMethod === "fixed"
+                        ? "swap2.form.rates.fixed"
+                        : "swap2.form.rates.float"
+                    }
+                  />
+                </Text>
+              </Box>
+            )}
+          </Box>
+          <Box alignItems="flex-end" flex={1}>
+            {centralized ? (
+              <>
+                <FormattedVal
+                  inline
+                  fontSize={4}
+                  val={amount}
+                  currency={toCurrency}
+                  unit={toCurrency?.units[0]}
+                  showCode={true}
+                  color="palette.text.shade100"
                   fontWeight="600"
                 />
-              </Box>
-              <Text fontSize={3} color="palette.text.shade40">
-                <Trans
-                  i18nKey={
-                    value.tradeMethod === "fixed"
-                      ? "swap2.form.rates.fixed"
-                      : "swap2.form.rates.float"
-                  }
+                <CounterValue
+                  fontSize={3}
+                  inline
+                  currency={toCurrency}
+                  value={amount}
+                  disableRounding
+                  showCode
+                  color="palette.text.shade40"
                 />
+              </>
+            ) : (
+              <Text
+                fontSize={3}
+                color="palette.text.shade40"
+                style={{ width: "110px", textAlign: "right" }}
+              >
+                <Trans i18nKey={"swap.providers.noQuote"} />
               </Text>
-            </Box>
-          </Box>
-          <Box alignItems="flex-end" flex={1}>
-            <FormattedVal
-              inline
-              fontSize={4}
-              val={amount}
-              currency={toCurrency}
-              unit={toCurrency?.units[0]}
-              showCode={true}
-              color="palette.text.shade100"
-              fontWeight="600"
-            />
-            <CounterValue
-              fontSize={3}
-              inline
-              currency={toCurrency}
-              value={amount}
-              disableRounding
-              showCode
-              color="palette.text.shade40"
-            />
+            )}
```

Comment:
Your Rate component is getting quite complex and messy by trying to handle centralised and decentralised providers in the same place. Especially since both centralised and decentralised rates have different sets of props.

You should rather break it down and use composition to have 2 different end components, a `CentralisedRate` (or `CentralisedProvider`) and a `DecentralisedRate` (or `DecentralisedProvider`, because for DEX we don't actually have rate).
You could have a generic `Rate` component for all the common logic / style and have the centralised and decentralised ones being specialised version of this generic `Rate` one.

cf. https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
Totally agree. Done.